pkg/reconciler/reconcile.go

```diff
@@ -96,9 +88,12 @@ func (r *GenericDeploymentReconciler) ReconcileAdapter(ctx context.Context, ab A
 
 	desiredAdapter := ab.BuildAdapter(rcl, sinkURI)
 
-	saOwners, err := ab.RBACOwners(rcl)
-	if err != nil {
-		return fmt.Errorf("listing ServiceAccount owners: %w", err)
+	saOwners := []kmeta.OwnerRefable{rcl}
+	if !v1alpha1.WantsOwnServiceAccount(rcl) {
+		var err error
+		if saOwners, err = rbacOwners(r.OwnersLister(rcl.GetNamespace())); err != nil {
+			return fmt.Errorf("listing ServiceAccount owners: %w", err)
+		}
 	}
```

Comment:
... and we use that generic lister to fetch the list of OwnerRefables, instead of requiring reconcilers to implement a `RBACOwners)()` method.